src/client/Microsoft.Identity.Client/AppConfig/AbstractApplicationBuilder.cs

```diff
@@ -528,6 +528,11 @@ public T WithTelemetryClient(params ITelemetryClient[] telemetryClients)
 
             if ((telemetryClients != null) && telemetryClients.Any())
             {
+                foreach (var telemetryClient in telemetryClients)
```

Comment:
The `params` will automatically create a non-null array for you. It will have 0 elements if the user does `WithTelemetryClient()`.